### PR TITLE
[Diagnostics] Filter operators if all their arguments are holes

### PR DESCRIPTION
**Pull request by xedin**
Restrict filtering in `simplifyAppliedOverloadsImpl` to disable
overload set for operators only, because other calls e.g. regular
functions or subscripts could be filtered on labels and are less
overloaded.

Filtering non-operator calls could also lead to incorrect diagnostics
because first choice could have all sorts of different issues e.g.
incorrect labels and number of parameters.

Resolves: rdar://55369704

### CODE REVIEW

**Comment by xedin** [comment]
@swift-ci please smoke test

File: lib/Sema/ConstraintSystem.cpp [hunk 1]
```diff
@@ -5689,3 +5689,18 @@ TypeVarBindingProducer::getDefaultBinding(Constraint *constraint) const {
              ? binding.withType(OptionalType::get(type))
              : binding;
 }
+
+bool constraints::isOperatorDisjunction(Constraint *disjunction) {
```
Comment (hborla) [inline lib/Sema/ConstraintSystem.cpp:5693]: This looks really similar to the code in `isOperatorBindOverload` in CSSolver.cpp. Could this new function subsume the old one?

**Review by hborla — APPROVED** [review]

File: lib/Sema/ConstraintSystem.cpp (hunk 1 above)
Comment (xedin) [reply lib/Sema/ConstraintSystem.cpp:5693]: heh, I suspected something like that but just couldn't find it...

**Comment by xedin** [comment]
@swift-ci please smoke test Windows platform

File: lib/Sema/ConstraintSystem.cpp (hunk 1 above)
Comment (xedin) [reply lib/Sema/ConstraintSystem.cpp]: Fixed by https://github.com/apple/swift/pull/36911.

**Comment by xedin** [comment]
@swift-ci please smoke test

**Pull request merged**